EssentialCSharp.Web/Areas/Identity/Pages/Account/Login.cshtml.cs

```diff
@@ -77,6 +78,8 @@ public async Task<IActionResult> OnPostAsync(string? returnUrl = null)
             if (foundUser is not null)
             {
                 result = await signInManager.PasswordSignInAsync(foundUser, Input.Password, Input.RememberMe, lockoutOnFailure: true);
+                // Call the referral service to get the referral ID and set it onto the user claim
+                _ = await referralService.GetReferralIdAsync(foundUser);
```

Comment:
I would never expect the result of any method named `Get...` to just be thrown away. If this is not a pure method, it probably needs a more appropriate name

Reply:
Updated to `EnsureReferralIdAsync`, I think this communicates that it will create it if it doesn't exist